forge-gui/src/main/java/forge/player/HumanCostDecision.java

```diff
@@ -382,9 +411,15 @@ private PaymentDecision exileFromMiscZone(final CostExile cost, final int nNeede
 
         final List<ZoneType> origin = Lists.newArrayList(cost.from);
         final CardCollection exiled = new CardCollection();
+        final String required = sharedType ? " (must share a card type)" : "";
 
-        final List<Card> chosen = controller.chooseCardsForZoneChange(ZoneType.Exile, origin, ability, typeList, mandatory ? nNeeded : 0,
-                nNeeded, null, cost.toString(), null);
+
+        final List<Card> chosen = controller.chooseCardsForZoneChange(ZoneType.Exile, origin, ability, typeList,
+                mandatory ? nNeeded : 0, nNeeded, null, cost.toString(nNeeded) + required,
+                null);
+        if (sharedType) {
+            if (!chosen.get(1).sharesCardTypeWith(chosen.get(0))) return null;
```

Comment:
Maybe human interface for "choose two with same type" (Eye of Ojer Taq) would be better to filter remaining options after first is chosen instead of just letting them choose any two and returning null if they goof?